beakerlib/index.js

```diff
@@ -1,56 +1,64 @@
-    const procTable = {};
-    for (let i = 0; i < val.length;) {
-        const proc = {};
-        // Key
-        proc.key = web3.toHex(val[i]); i++;
-        if (proc.key == "0x0") break;
-        // KeyIndex
-        proc.keyIndex = web3.toHex(val[i]); i++;
-        // Location
-        proc.location = web3.toHex(val[i]); i++;
-        // Capabilities
-        proc.caps = [];
-        const nCaps = val[i].toNumber(); i++;
-        for (let j = 0; j < nCaps; j++) {
-            const cap = {};
-            const length = web3.toHex(val[i]); i++;
-            cap.type = web3.toHex(val[i]); i++;
-            // (length - 1) as the first value is the length
-            cap.values = [];
-            for (let k = 0; k < (length-1); k++) {
-                cap.values.push(web3.toHex(val[i])); i++;
+class ProcedureTable {
+    constructor(procTable) {
+        this.procedures = procTable;
+    }
+    // get procedures() {
+    //     return this.procTable;
+    // }
+    static parse(val) {
+        const procTable = {};
+        for (let i = 0; i < val.length;) {
+            const proc = {};
+            // Key
+            proc.key = web3.toHex(val[i]); i++;
+            if (proc.key == "0x0") break;
+            // KeyIndex
+            proc.keyIndex = web3.toHex(val[i]); i++;
+            // Location
+            proc.location = web3.toHex(val[i]); i++;
+            // Capabilities
+            proc.caps = [];
+            const nCaps = val[i].toNumber(); i++;
+            for (let j = 0; j < nCaps; j++) {
+                const cap = {};
+                const length = web3.toHex(val[i]); i++;
+                cap.type = web3.toHex(val[i]); i++;
+                // (length - 1) as the first value is the length
+                cap.values = [];
+                for (let k = 0; k < (length-1); k++) {
+                    cap.values.push(web3.toHex(val[i])); i++;
+                }
+                proc.caps.push(cap);
             }
-            proc.caps.push(cap);
+            procTable[proc.key] = proc;
         }
-        procTable[proc.key] = proc;
+        return new ProcedureTable(procTable);
     }
-    return procTable;
-}
-
-exports.printProcedureTable = printProcedureTable;
-function printProcedureTable(procTable) {
-    for (const procKey of Object.keys(procTable)) {
-        const proc = procTable[procKey];
-        // Print key
-        console.log(`Key: ${proc.key}`);
-        // Print keyIndex
-        console.log(`  KeyIndex: ${proc.keyIndex}`);
-        // Print location
-        console.log(`  Location: ${proc.location}`);
-        // Print Capabilities
-        console.log(`  Capabilities(${proc.caps.length} keys)`);
-        for (const i in proc.caps) {
-            const cap = proc.caps[i];
-            console.log(`    Capability[${i}]: Type: ${cap.type}`);
-            for (const j in cap.values) {
-                console.log(`      ${j}: ${cap.values[j]}`)
-
+    static stringify(procTable) {
```

Comment:
This should be a instance method

Reply:
Other than that looks good to merge.

Reply:
I had that, but switched to this to try and match convention rather than overwrite object properties. It matches the `JSON` style. What's the prevailing convention?

Reply:
You're not overwriting object properties in this case, you're adding a function into the top of the prototype chain, so it gets chosen first. [You only overwrite if you do `Object.propotype.foo = ...`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)



So writing a `#toString` method would be the prevailing convention.